common/CMakeLists.txt

```
@@ -13,99 +13,120 @@
 
-ADD_LIBRARY(${BLOCK_STORE_LIB_NAME} STATIC packages/block_store/lmdb_block_store.cpp)
+  ADD_LIBRARY(${U_COMMON_LIB_NAME} STATIC ${PROJECT_HEADERS} ${PROJECT_SOURCES})
 
-TARGET_INCLUDE_DIRECTORIES(${BLOCK_STORE_LIB_NAME} PRIVATE ${COMMON_PRIVATE_INCLUDE_DIRS})
-TARGET_INCLUDE_DIRECTORIES(${BLOCK_STORE_LIB_NAME} PUBLIC ${SGX_SDK}/include)
+  TARGET_INCLUDE_DIRECTORIES(${U_COMMON_LIB_NAME} PRIVATE ${COMMON_INCLUDE_DIRS})
+  TARGET_INCLUDE_DIRECTORIES(${U_COMMON_LIB_NAME} PRIVATE ${SGX_SDK}/include)
 
-TARGET_COMPILE_OPTIONS(${BLOCK_STORE_LIB_NAME} PRIVATE ${COMMON_CXX_FLAGS} ${OPENSSL_CFLAGS})
-TARGET_COMPILE_DEFINITIONS(${BLOCK_STORE_LIB_NAME} PRIVATE "-D_UNTRUSTED_=1")
+  TARGET_COMPILE_OPTIONS(${U_COMMON_LIB_NAME} PRIVATE ${COMMON_CXX_FLAGS} ${OPENSSL_CFLAGS})
 
-option(BLOCK_STORE_DEBUG "Debug logging for block store operations" OFF)
-if (BLOCK_STORE_DEBUG)
-    TARGET_COMPILE_DEFINITIONS(${UNTRUSTED_LIB_NAME} PRIVATE "-DBLOCK_STORE_DEBUG=1")
-endif()
+  TARGET_COMPILE_DEFINITIONS(${U_COMMON_LIB_NAME} PRIVATE "_UNTRUSTED_=1")
+ENDIF()
 
 ################################################################################
 # Trusted Common Library
 ################################################################################
+IF (BUILD_TRUSTED)
+  PROJECT(${T_COMMON_LIB_NAME} CXX)
 
-if(NOT UNTRUSTED_ONLY)
-	SET(TRUSTED_LIB_NAME tpdo-common)
-	PROJECT(${TRUSTED_LIB_NAME} CXX)
+  ADD_LIBRARY(${T_COMMON_LIB_NAME} STATIC ${PROJECT_HEADERS} ${PROJECT_SOURCES})
 
-	ADD_LIBRARY(${TRUSTED_LIB_NAME} STATIC ${PROJECT_HEADERS} ${PROJECT_GENERATED_IAS_SOURCES} ${PROJECT_SOURCES})
-
-	TARGET_INCLUDE_DIRECTORIES(${TRUSTED_LIB_NAME} PRIVATE ${COMMON_PRIVATE_INCLUDE_DIRS})
-	TARGET_INCLUDE_DIRECTORIES(${TRUSTED_LIB_NAME} PUBLIC ${SGX_SDK}/include)
-	TARGET_INCLUDE_DIRECTORIES(${TRUSTED_LIB_NAME} PUBLIC ${SGX_SSL}/include)
-	TARGET_INCLUDE_DIRECTORIES(${TRUSTED_LIB_NAME} PUBLIC ${SGX_SDK}/include/tlibc)
-	TARGET_INCLUDE_DIRECTORIES(${TRUSTED_LIB_NAME} PUBLIC ${SGX_SDK}/include/libcxx)
-
-	TARGET_COMPILE_OPTIONS(${TRUSTED_LIB_NAME} PRIVATE ${COMMON_CXX_FLAGS})
-	TARGET_COMPILE_OPTIONS(${TRUSTED_LIB_NAME} PRIVATE -nostdinc)
-	TARGET_COMPILE_OPTIONS(${TRUSTED_LIB_NAME} PRIVATE -nostdinc++)
-	TARGET_COMPILE_OPTIONS(${TRUSTED_LIB_NAME} PRIVATE -fno-builtin-printf)
+  TARGET_INCLUDE_DIRECTORIES(${T_COMMON_LIB_NAME} PRIVATE ${COMMON_INCLUDE_DIRS})
+  TARGET_INCLUDE_DIRECTORIES(${T_COMMON_LIB_NAME} PRIVATE ${SGX_SDK}/include)
+  TARGET_INCLUDE_DIRECTORIES(${T_COMMON_LIB_NAME} PRIVATE ${SGX_SSL}/include)
+  TARGET_INCLUDE_DIRECTORIES(${T_COMMON_LIB_NAME} PRIVATE ${SGX_SDK}/include/tlibc)
+  TARGET_INCLUDE_DIRECTORIES(${T_COMMON_LIB_NAME} PRIVATE ${SGX_SDK}/include/libcxx)
 
+  TARGET_COMPILE_OPTIONS(${T_COMMON_LIB_NAME} PRIVATE ${COMMON_CXX_FLAGS})
+  TARGET_COMPILE_OPTIONS(${T_COMMON_LIB_NAME} PRIVATE -nostdinc)
+  TARGET_COMPILE_OPTIONS(${T_COMMON_LIB_NAME} PRIVATE -nostdinc++)
+  TARGET_COMPILE_OPTIONS(${T_COMMON_LIB_NAME} PRIVATE -fno-builtin-printf)
 endif()
```

Yes, this is a total nit.

```suggestion
ENDIF()
```